app/services/remote_job_handler.rb

```diff
@@ -66,6 +69,43 @@ def set_submitted_to_if_necessary(job)
     end
   end
 
+  def execute_local_pre_process(job)
+    return unless job.executable.local_pre_process_script.present?
+    Dir.chdir( job.dir ) {
```

Comment:
Local pre process runs on public/Result_development/`executable.id`. It means that some files may be overwritten if remote jobs create files with the same name. Is it a desired specification?